modules/serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/ServiceRegistryJpaImpl.java

```diff
@@ -2196,10 +2315,13 @@ public List<ServiceRegistration> getServiceRegistrationsByLoad(String serviceTyp
     EntityManager em = null;
     try {
       em = emf.createEntityManager();
-      SystemLoad loadByHost = getHostLoads(em);
+      SystemLoad loads = getHostLoads(em);
       List<HostRegistration> hostRegistrations = getHostRegistrations();
       List<ServiceRegistration> serviceRegistrations = getServiceRegistrationsByType(serviceType);
-      return getServiceRegistrationsByLoad(serviceType, serviceRegistrations, hostRegistrations, loadByHost);
+      // if enabled use the hardware loads instead
+      if (hardwareLoadEnabled)
+        loads = getSystemHardwareLoad();
+      return getServiceRegistrationsByLoad(serviceType, serviceRegistrations, hostRegistrations, loads);
```

Comment:
As the loads variable gets overwritten, you could use the if statement to retrieve only the used load.